api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -74,5 +90,22 @@
    *
    * @return this for method chaining
    */
-  OverwriteFiles validateAddedFiles();
+  OverwriteFiles validateAddedFilesMatchRowFilter();
+
+  /**
+   * Enables validation of files that are added concurrently.
```

Comment:
I'd like to add something about the type of validation. Does something like this make sense?

> Enables validation that files added concurrently do not conflict with this commit's operation.